ui/lib/core/addon/components/download-csv.js

```diff
@@ -17,14 +17,25 @@ import { action } from '@ember/object';
  * @param {string} fileName - Custom name for the downloaded file
  *
  */
+
+// Sample of CSV format: each row needs to be \n separated, and each column separated by a comma
+// 'Namespace path,Authentication method,Total clients,Entity clients,Non-entity clients\n
+//  namespacelonglonglong4/,,191,171,20\n
+//  namespacelonglonglong4/,auth/method/uMGBU,35,20,15\n'
+
 class DownloadCsvComponent extends Component {
   @action
   downloadCsv() {
     let hiddenElement = document.createElement('a');
     hiddenElement.setAttribute('href', 'data:text/csv;charset=utf-8,' + encodeURI(this.args.csvData));
     hiddenElement.setAttribute('target', '_blank');
     hiddenElement.setAttribute('download', this.args.fileName || 'vault-data.csv');
-    hiddenElement.click();
+    try {
+      hiddenElement.click();
+      this.args.handleDownloadSuccess();
+    } catch (e) {
+      console.log(e);
```

Comment:
Added this `try...catch` to play with ways to close the modal after downloading the file. Will probably want to handle the error differently than console logging if we keep this...